src/api/tasks.js

```diff
@@ -42,14 +42,17 @@ const buildTaskHierarchy = tasks => {
   const taskMap = {};
   const rootTaskGids = [];
   for (let i = 0; i < tasks.length; i++) {
-    const t = tasks[i];
-    taskMap[t.gid] = Object.assign(t, { subtask_gids: [] });
+    const t1 = tasks[i];
+    taskMap[t1.gid] = Object.assign(t1, { subtask_gids: [] });
+  }
+  for (let i = 0; i < tasks.length; i++) {
+    const t2 = tasks[i];
 
-    if (!t.parent_task_gid) {
-      rootTaskGids.push(t.gid);
+    if (!t2.parent_task_gid) {
+      rootTaskGids.push(t2.gid);
     } else {
       // Add t to its parent's subtasks
-      taskMap[t.parent_task_gid].subtask_gids.push(t.gid);
+      taskMap[t2.parent_task_gid].subtask_gids.push(t2.gid);
```

Comment:
Glad you caught this. We _should_ be able to replace the `GET` to `/postgrest/tasks` (see `getTasksReq`) with a `POST` to `/postgrest/rpc/ `, to call `subtasks_recursive_flat` (see https://github.com/PursuanceProject/pursuance/blob/820af875c9aa7743a974c38c23ec7536c7bcf7c2/db/sql/migration0003.sql#L3 ), which orders the tasks such that we'd only need to iterate over the resulting task array once rather than twice.

...but it might actually be faster to just do what you've done here -- iterate twice -- rather than do a more complex DB query to get things in the right order; I'm not sure.